src/main/java/com/spring/taskmanger/filters/AuthTokenFilter.java

```diff
@@ -0,0 +1,80 @@
+package com.spring.taskmanger.filters;
+
+import com.spring.taskmanger.dao.TokenRepository;
+import com.spring.taskmanger.dao.UserRepository;
+import com.spring.taskmanger.security.CustomUserDetails;
+import com.spring.taskmanger.security.CustomUserDetailsService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+//@Component
+public class AuthTokenFilter extends OncePerRequestFilter {
+    @Autowired
+    private JwtUtil jwtUtils;
+
+    @Autowired
+    private CustomUserDetailsService userDetailsService ;
+
+    @Autowired
+    private UserRepository UserRepository;
+
+    @Autowired
+    private TokenRepository tokenRepository;
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthTokenFilter.class);
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        try {
+            String jwt = parseJwt(request);
+            if (jwt != null && jwtUtils.validateJwtToken(jwt) ) {
+                String username = jwtUtils.getUserNameFromJwtToken(jwt);
+                logger.info("inside AuthTokenFilter username: " + username);
+                CustomUserDetails userDetails = (CustomUserDetails) userDetailsService.loadUserByUsername(username);
```

Comment:
Did you depend on the user name to get the user details from the database?

Reply:
Yes , i used users email as the username 
and it is unique

Reply:
Okay, but as we have a name field in the user, it leads to misunderstanding by the one who read the code.
Another thing why don't use user-id instead of email?

Reply:
commonly we use email address for logging in , why to use user-id ? 
and how i can make it more understandable  ?

Reply:
It seems I misunderstand something here, my note was to put the user id in the token instead of the email.
To make it understandable, maybe you can change the method neme, but it's okay for now, it's not a big deal.

Reply:
I didn't mean to use the user id for login.